app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -254,16 +270,22 @@ def fog_credentials
         :openstack_api_key  => password,
         :openstack_username => user,
         :openstack_auth_url => url_for_fog,
-        :openstack_tenant   => tenant,
-        :openstack_identity_endpoint => url_for_fog,
-        :openstack_user_domain       => domain,
-        :openstack_endpoint_type     => "publicURL"
+        :openstack_identity_endpoint => url_for_fog
+        #:openstack_endpoint_type     => "publicURL"
       }.tap do |h|
-        if tenant
-          h[:openstack_domain_name] = domain
-          h[:openstack_project_name] = tenant
+        if tenant != ''
+          if identity_version == 2
+            h[:openstack_tenant] = tenant
+          else
+            h[:openstack_project_name] = tenant
+          end
         end
+        h[:openstack_user_domain] = domain if domain != ''
```

Comment:
if coming from API, all these values can be nil right? should we change it here and next 3 lines to `.present?`